pkg/storage/chunk/chunk_store_utils.go

```diff
@@ -105,26 +123,60 @@ type decodeResponse struct {
 // NewChunkFetcher makes a new ChunkFetcher.
 func NewChunkFetcher(cacher cache.Cache, cacheStubs bool, schema SchemaConfig, storage Client) (*Fetcher, error) {
 	c := &Fetcher{
-		schema:         schema,
-		storage:        storage,
-		cache:          cacher,
-		cacheStubs:     cacheStubs,
-		decodeRequests: make(chan decodeRequest),
+		schema:              schema,
+		storage:             storage,
+		cache:               cacher,
+		cacheStubs:          cacheStubs,
+		decodeRequests:      make(chan decodeRequest),
+		maxAsyncConcurrency: 16,
+		maxAsyncBufferSize:  1000,
```

Comment:
Should this not be configurable?

Reply:
done, thanks for review。
add two config in pkg/storage/chunk/cache/cache.go:46
`
	// MaxAsyncConcurrency specifies the maximum number of SetAsync goroutines.
	MaxAsyncConcurrency int `yaml:"max_async_concurrency"`
	// MaxAsyncBufferSize specifies the queue buffer size for SetAsync operations.
	MaxAsyncBufferSize int `yaml:"max_async_buffer_size"`
`

![image](https://user-images.githubusercontent.com/9583245/148733550-0683d0f6-64f0-4585-9c49-944b173e989a.png)